Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs

```diff
@@ -170,20 +173,20 @@ private static bool IsPythonSite(string projectPath)
             return PythonSiteEnabler.LooksLikePython(projectPath);
         }
 
-        private ISiteBuilder ResolveProject(string repositoryRoot, IDeploymentSettingsManager perDeploymentSettings, IFileFinder fileFinder, bool tryWebSiteProject = false, SearchOption searchOption = SearchOption.AllDirectories)
+        private ISiteBuilder ResolveProject(string repositoryRoot, IDeploymentSettingsManager perDeploymentSettings, IFileFinder cashedFileFinder, IFileFinder fileFinder, bool tryWebSiteProject = false, SearchOption searchOption = SearchOption.AllDirectories)
         {
-            return ResolveProject(repositoryRoot, repositoryRoot, perDeploymentSettings, fileFinder, tryWebSiteProject, searchOption, specificConfiguration: false);
+            return ResolveProject(repositoryRoot, repositoryRoot, perDeploymentSettings, cashedFileFinder, fileFinder, tryWebSiteProject, searchOption, specificConfiguration: false);
         }
 
-        private ISiteBuilder ResolveProject(string repositoryRoot, string targetPath, IDeploymentSettingsManager perDeploymentSettings, IFileFinder fileFinder, bool tryWebSiteProject, SearchOption searchOption = SearchOption.AllDirectories, bool specificConfiguration = true)
+        private ISiteBuilder ResolveProject(string repositoryRoot, string targetPath, IDeploymentSettingsManager perDeploymentSettings, IFileFinder cachedFileFinder, IFileFinder fileFinder, bool tryWebSiteProject, SearchOption searchOption = SearchOption.AllDirectories, bool specificConfiguration = true)
```

Comment:
Just reading through the code, it's not very clear why we need two different `IFileFinder` passed in, and what each is for. Maybe needs a few comments :)